### ISSUE TITLE
Type of pre hook callback

### ISSUE THREAD

**Issue by lafeuil** [typegoose/typegoose#90]
## Versions
- Typegoose(NPM): 6.0.4
  
## What is the Problem?

I think the type of pre hook callback are not good. It is defined by `EmptyVoidFn`. So, it is not possible to call the next callback with an error.

## Code Example
```ts
@pre<Car>('save', function(next) {
   next(new Error('driver not found')); // Typescript ERROR : Expected 0 arguments, but got 1
})
export class Car {}
```

## Do you know *why* it happenes?

It is :
```ts
type PreFnWithDT<T> = (this: DocumentType<T>, next: EmptyVoidFn) => void;
```
but it may be :
```ts
type PreFnWithDT<T> = (this: DocumentType<T>, next: HookNextErrorFn) => void;
```

**Comment by JakeAi** [comment]
I noticed this as well.

**Comment by hasezoey** [comment]
for now you can just throw in the functions, otherwise it will be fixed in 6.1.0 (31328dafba61363aba5e8be3d72787d6bbea19fe)

otherwise, if anyone finds wrong types in the hooks again, please open a new issue - hook types are confusing because they change with context
